# Nonlinear Transformation Models

Download the EDSAL.csv file and upload it to a dataframe (in R). The three variables are Education, Experience and Salary.

Run 4 linear regressions using the lm function in R. (note – you have to use the natural log; the default 'log()' function in R)

- Linlin: Use Salary as the dependent variable and Experience as the independent variable.
- Linlog: Use Salary as the dependent variable and log(Experience) as the independent variable.
- Loglin: Use log(Salary) as the dependent variable and Experience as the independent variable.
- Loglog: Use log(Salary) as the dependent variable and log(Experience) as the independent variable.

In [1]:
#load required library
library(ggplot2) 
library("tidyverse")
library(Ecdat)
library(ISLR)
df<-read.csv("EDSAL.csv")

Warning message:
"package 'ggplot2' was built under R version 3.4.4"Warning message:
"package 'tidyverse' was built under R version 3.4.4"-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
v tibble  1.4.2     v purrr   0.2.5
v tidyr   0.8.1     v dplyr   0.7.6
v readr   1.1.1     v stringr 1.3.1
v tibble  1.4.2     v forcats 0.3.0
Warning message:
"package 'tibble' was built under R version 3.4.4"Warning message:
"package 'tidyr' was built under R version 3.4.4"Warning message:
"package 'readr' was built under R version 3.4.4"Warning message:
"package 'purrr' was built under R version 3.4.4"Warning message:
"package 'dplyr' was built under R version 3.4.4"Warning message:
"package 'stringr' was built under R version 3.4.4"Warning message:
"package 'forcats' was built under R version 3.4.4"-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()
Warning mess

In [2]:
df<- df %>%
    mutate(ln_Salary=log(Salary)) %>%
    mutate(ln_Experience=log(Experience))
head(df)

Warning message:
"package 'bindrcpp' was built under R version 3.4.4"

Education Experience Salary    ln_Salary ln_Experience
1 HS         2          34.43222 3.538993  0.6931472    
2 HS        14          59.15737 4.080201  2.6390573    
3 HS        36         113.27017 4.729776  3.5835189    
4 HS        16          69.14739 4.236240  2.7725887    
5 HS        36         106.01556 4.663586  3.5835189    
6 HS        33          58.63413 4.071317  3.4965076

In [3]:
linlin<-lm(formula=Salary~Experience,data=df)
linlog<-lm(formula=Salary~ln_Experience,data=df)
loglin<-lm(formula=ln_Salary~Experience,data=df)
loglog<-lm(formula=ln_Salary~ln_Experience,data=df)

In [6]:
summary(linlin)$r.squared 
summary(linlog)$r.squared 
summary(loglin)$r.squared 
summary(loglog)$r.squared 
linlin$coefficients['Experience']
linlog$coefficients['ln_Experience']
loglin$coefficients['Experience']
loglog$coefficients['ln_Experience']


[1] 0.7218044

[1] 0.5857976

[1] 0.7433722

[1] 0.7252159

Experience 
  3.095937

ln_Experience 
     34.98505

Experience 
 0.0370868

ln_Experience 
    0.4594901

- ** Which of the 4 fitted models has the lowest R-square value? 0.5858 linlog ** 

- ** Which of the 4 fitted models has the highest R-square value? 0.7434 loglin **

- ** What is the interpretation of the slope coefficient for the Linlin model ? <br> Increasing Experience by 1 unit leads to 3.0959 units increase in Salary** 

- ** What is the interpretation of the slope coefficient for the Linlog model?<br> Increasing Experience by 1% leads to 0.01*34.985 units increase in Salary **

- ** What is the interpretation of the slope coefficient for the Loglin model ?<br> Increasing Experience by 1 unit leads to (0.037087 *100) % increase in Salary **
- ** Increasing Experience by 1 unit leads to (0.037087 *100) % increase in Salary ? <br> Increasing Experience by 1% leads to 0.45949% increase in Salary **